PD-PiotrKurzydym/praca-domowaJS-nr1-Piotr-Kurzydym.js

```diff
@@ -0,0 +1,605 @@
+      e) Have fun :)
+      Na przykład:
+      Dla osoby: 
+      {
+          firstName: 'Bartolomeo',
+          lastName: 'Lozano'
+      }
+      powinniśmy uzyskać nickname Rabona
+      Hints:
+      - mając zmienną name = 'Bart'
+        name.split('') => ['B', 'a', 'r', 't'] - tworzymy tablicę liter ze stringa
+        ['B', 'a', 'r', 't'].join('') => 'Bart' - odwracamy ten proces
+      - Na tablicy możemy użyć metody reverse()
+      - Na stringach czy pojedynczych literkach możemy używać metod toLowerCase(), toUpperCase()
+  */
+
+console.log(`Zadanie 2.a)`);
+const people2 = [
+  {
+    firstName: "Bartolomeo",
+    lastName: "Lozano",
+    nickname: "Rabona",
+    introduceYourself() {
+      console.log(
+        `Cześć jestem ${this.firstName} ${this.lastName}, ale w szkole mówią na mnie [${this.nickname}]`
+      );
+    },
+  },
+  {
+    firstName: "Piotrulo",
+    lastName: "Kurzydymulo",
+    nickname: "Kurzy",
+    introduceYourself() {
+      console.log(
+        `Cześć jestem ${this.firstName} ${this.lastName}, ale w szkole mówią na mnie [${this.nickname}]`
+      );
+    },
+  },
+];
+people2[0].introduceYourself();
+console.log(`---`);
+
+console.log(`Zadanie 2.b)`);
+people2.forEach(function (item) {
+  item.introduceYourself();
+});
+console.log(`---`);
+
+/* 
+//  DANE WEJŚCIOWE
+// 2.
+// a) Do każdego obiektu dodaj funkcję introduceYourself, która za pomocą słówka this wyświetli w konsoli tekst powitalny.
+// b) za pomocą pętli forEach, wywołaj funkcję powitalną dla każdego elementu tablicy. W rezultacie na ekranie powinien
+// pojawić się tekst powitalny dla każdej osoby w tablicy
+      Oczywiście tekst powinien wyświetlić się dopiero po wywołaniu funkcji.
+      Dla powyższego przykładu tekst powinien wyglądać w następujący sposób:
+      "Cześć jestem Bartolomeo Lozano, ale w szkole mówią na mnie [Rabona]"
+      Natomiast wywołanie funkcji: people[0].introduceYourself()
+      Obiekt z przykładu powinien wyglądać w ten sposób
+      {
+          firstName: "Bartolomeo",
+          lastName: "Lozano",
+          nickname: "Rabona",
+          introduceYourself: // tutaj ma się znajdować funkcja
+      },
+      b) za pomocą pętli forEach, wywołaj funkcję powitalną dla każdego elementu tablicy. W rezultacie na ekranie powinien
+      pojawić się tekst powitalny dla każdej osoby w tablicy
+
+      Hints:
+      - nie używaj w tym zadaniu funkcji strzałkowej, ponieważ słówko this Ci nie zadziała i nie będziesz miał(a)
+      dostępu do this.firstName lastName i nickname
+      - postaraj się zdefiniować funkcję powitalną tylko raz (nie rób tego w pętli, ani funkcji map)
+      
+
+  */
+
+//  DANE WEJŚCIOWE
+console.log(`Zadanie 3.`);
+const colors = ["red", "green", "yellow", "blue", "pink", "orange"];
+const people3 = [
+  {
+    firstName: "Bartolomeo",
+    lastName: "Lozano",
+    nickname: "Rabona",
+    getFavouriteColor(a) {
```

Comment:
Tu jest podobna sytuacja jak wyżej. Zadeklarowanie funkcji poza obiektem pozwoliło by na jej używanie w różnych miejscach w kodzie. Zapisując ją w ten sposób ograniczasz się tylko do zakresu obiektu. Dobrą praktyką jest też konkretne nazywanie zmiennych i parametrów. Pozwala to łatwiej czytać kod.